### PR TITLE
more changes in the base classes and added the Employee resource

### PR DESCRIPTION
**Pull request by reconeazaretzky**
created a base crud service
updated the code to use it
added employees

### CODE REVIEW

File: src/main/java/com/entropyteam/entropay/employees/common/BaseRepository.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.entropyteam.entropay.employees.common;
```
Comment (arael82) [inline src/main/java/com/entropyteam/entropay/employees/common/BaseRepository.java:1]: I'd move this interface to something like com.entropyteam.entropay.common.repository, in order to keep it outside employees or any other module-specific package.

File: src/main/java/com/entropyteam/entropay/employees/common/BaseService.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package com.entropyteam.entropay.employees.common;
```
Comment (arael82) [inline src/main/java/com/entropyteam/entropay/employees/common/BaseService.java:1]: I'd move this interface to something like com.entropyteam.entropay.common.services, in order to keep it outside employees or any other module-specific package.

File: src/main/java/com/entropyteam/entropay/employees/common/CrudService.java [hunk 3]
```diff
@@ -3,15 +3,15 @@
 import java.util.List;
 import java.util.Optional;
 
-public interface CrudService<T, K> {
+public interface CrudService<DTO, Key> {
```
Comment (arael82) [inline src/main/java/com/entropyteam/entropay/employees/common/CrudService.java:6]: I'd move this interface to something like com.entropyteam.entropay.common.services, in order to keep it outside employees or any other module-specific package.

File: src/main/java/com/entropyteam/entropay/employees/common/BaseRepository.java (hunk 1 above)
Comment (reconeazaretzky) [reply src/main/java/com/entropyteam/entropay/employees/common/BaseRepository.java:1]: ideally should be a common artifact, but we can do a common package for now

File: src/main/java/com/entropyteam/entropay/employees/clients/models/Client.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
 import lombok.Setter;
 
 @Entity(name = "Client")
-@Table(name = "client")
+@Table(name = "client", schema = "employees")
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/clients/models/Client.java:15]: i thought in use 1 db per microservice and use public schema for each db

File: src/main/java/com/entropyteam/entropay/employees/entropist/models/Employee.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
+package com.entropyteam.entropay.employees.entropist.models;
+
+import java.time.LocalDate;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import com.entropyteam.entropay.employees.common.BaseEntity;
+import com.entropyteam.entropay.employees.entropist.dtos.EmployeeDto;
+
+@Entity(name = "Employee")
+@Table(name = "employee", schema = "employees")
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/entropist/models/Employee.java:10]: Same here schema is `public`

**Pull request merged**